software-documentation: add section, embrace manpages (#545)
void-linux/void-docs: src/config/package-documentation/index.md

# Package documentation

The most common media for documentation in Void Linux are [manual
pages](./man.md).

Many packages contain documentation in other formats, like HTML. This
documentation can usually be found in a `/usr/share/doc/<package>` directory.

More extensive documentation may be split into separate `*-doc` packages, such
as `julia-doc`. This is often the case for programming languages, databases and
big software libraries.

In addition to documentation provided by upstream projects, packages may also
contain description of initial setup or usage specific to Void, provided by

> distro contributors. It will be located in

Reviewer: My only nit is "distro" -> "distribution", which is more widespread in docs. Otherwise LGTM

Author: Done.

> In addition to documentation provided by upstream authors, packages may also
> contain some provided by Void project. This typically describes initial setup,
> or usage specific to Void. Location of it is a
> `/usr/share/doc/<package>/README.voidlinux`.

Reviewer: ```suggestion
In addition to documentation provided by upstream authors, packages may also
contain additional documentation provided by Void Linux (or "the Void project"). This typically describes initial setup,
or usage specific to Void. It can be located in
`/usr/share/doc/<package>/README.voidlinux`.
```

Author: "In addition ... additional" sounds poor.

"Void project" is already a term in docs, do we want to be that verbose here?

What about "It will" rather that "It can"?

Reviewer: > What about "It will" rather that "It can"?

Yes

> "Void project" is already a term in docs, do we want to be that verbose here?

It's ok to use it, but then it needs "the" before it (what I suggested in parentheses).

> "In addition ... additional" sounds poor.

Hmm, true. Maybe "In addition to documentation provided by upstream projects, packages may also
contain other documentation provided" ? (I think switching from "authors" to "upstream" should happen either way).

Reviewer: Actually, I don't like "other" either.

Author: reworded
